R/parameters.R

```diff
@@ -150,25 +150,33 @@ unk_check <- function(x) {
 #' @export
 print.parameters <- function(x, ...) {
   x <- tibble::as_tibble(x)
+
   cat("Collection of", nrow(x), "parameters for tuning\n\n")
 
   print_x <- x %>% dplyr::select(identifier = id, type = name, object)
-  print_x$object <- purrr::map_chr(print_x$object, dplyr::type_sum)
+  print_x$object <-
+    purrr::map_chr(
+      print_x$object,
+      ~if (all(is.na(.x))) {"missing"} else {dplyr::type_sum(.x)}
+    )
+
   print.data.frame(print_x, row.names = FALSE)
   cat("\n")
 
   null_obj <- map_lgl(x$object, ~ all(is.na(.x)))
-  num_missing <- sum(null_obj)
-  if (num_missing > 0) {
-    if (num_missing == 1) {
-      cat("One needs a `param` object: '", x$identifier[null_obj], "'\n\n", sep = "")
-    } else {
-      cat(
-        "Several need `param` objects: ",
-        paste0("'", x$identifier[null_obj], "'", collapse = ", "),
-        "\n\n"
-      )
-    }
+
+  if (any(null_obj)) {
+    needs_param <- print_x$identifier[null_obj]
+
+    param_descs <- combine_words(print_x$identifier[null_obj], before = "`")
+
+    plural <- length(needs_param) != 1
+
+    rlang::inform(
+      glue::glue("The parameter{if (plural) 's' else ''} {param_descs} ",
+                 "{if (plural) {'need `param` objects'} else {'needs a `param` object'}}. ",
+                 "\nSee `vignette('dials')` to learn more.")
```

Comment:
Is there a better place to point than `vignette('dials')`?

Reply:
There is https://www.tidymodels.org/learn/develop/parameters/ which is specifically about creating new `param` objects. Do you think that gives enough context? If not, the Get Started vignette (i.e. `vignette('dials')`) does link out to the specific article so we could keep using the vignette as the starting point.